crates/storage/provider/src/providers/static_file/metrics.rs

```diff
@@ -80,6 +80,28 @@ impl StaticFileProviderMetrics {
                 .record(duration.as_secs_f64());
         }
     }
+
+    pub(crate) fn record_segment_operations(
+        &self,
+        segment: StaticFileSegment,
+        operation: StaticFileProviderOperation,
+        count: u64,
+        duration: Option<Duration>,
+    ) {
+        self.segment_operations
+            .get(&(segment, operation))
+            .expect("segment operation metrics should exist")
+            .calls_total
+            .increment(count);
+
+        if let Some(duration) = duration {
+            self.segment_operations
+                .get(&(segment, operation))
+                .expect("segment operation metrics should exist")
+                .write_duration_seconds
+                .record(duration.as_secs_f64() / count as f64);
+        }
```

Comment:
note for the future, let's remove the `as_str` way of doing metrics for this type

Reply:
just made an issue for this https://github.com/paradigmxyz/reth/issues/8758